airbyte-webapp/src/views/Connector/ServiceForm/components/AuthButtonFactory.tsx

```diff
@@ -0,0 +1,58 @@
+import { Button } from "../../../../components/base/Button";
+import { FormattedMessage } from "react-intl";
+import { ConnectorDefinition } from "../../../../core/domain/connector";
+import { isSourceDefinition } from "../../../../core/domain/connector/source";
+import { isDestinationDefinition } from "../../../../core/domain/connector/destination";
+
+const getButton = (
+  loading: boolean,
+  run: () => void,
+  selectedService?: ConnectorDefinition,
+  done?: boolean
+): JSX.Element => {
+  let definitionId = "";
+  if (selectedService && isSourceDefinition(selectedService)) {
+    definitionId = selectedService.sourceDefinitionId;
+  } else if (selectedService && isDestinationDefinition(selectedService)) {
+    definitionId = selectedService.destinationDefinitionId;
+  } else {
+    throw new Error(`Unrecognized service type. Service: ${selectedService}`);
+  }
+
+  switch (definitionId) {
+    case "253487c0-2246-43ba-a21f-5116b20a2c50": // google ads
```

Comment:
I think it's better to extract it to either a separate method or file and call it like `isGoogleOauth`, so if we add more google connectors - we will change that file only and won't change components implementation.

Reply:
Sounds good. I can remove the switch and do the suggested approach